src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -5343,8 +5343,11 @@ ActiveShardPlacementLists(List *taskList)
 
 
 /*
- * CompareShardPlacements compares two shard placements by their tuple oid; this
- * oid reflects the tuple's insertion order into pg_dist_placement.
+ * CompareShardPlacements compares two shard placements by placement id.
+ *
+ * We use this function to compare GroupShardPlacement objects too because
+ * placement id is a common field for GroupShardPlacement and ShardPlacement
+ * structs.
```

Comment:
no :)

this is something that C lets you do and can work by accident, but placementId will just be interpreted as a pointer offset